scrapy/downloadermiddlewares/ajaxcrawl.py

```diff
@@ -89,8 +88,7 @@ def _has_ajaxcrawlable_meta(text):
     if 'content' not in text:
         return False
 
-    text = _script_re.sub(u'', text)
-    text = _noscript_re.sub(u'', text)
-    text = html.remove_comments(html.replace_entities(text))
+    text = html.remove_tags_with_content(text, ('script', 'noscript'))
```

Comment:
Yup. Done.

script:
```py
import timeit

timed = timeit.timeit("""
texts = [
    '<html><head><meta name="fragment"  content="!"/></head><body></body></html>',
    "<html><head><meta name='fragment' content='!'></head></html>",
    '<html><head><!--<meta name="fragment"  content="!"/>--></head><body></body></html>',
    '<html></html>',
]

# [1] all of ajaxcrawl func
#for text in texts:
#    _has_ajaxcrawlable_meta(text)

# [2] w3lib
#for text in texts:
#    remove_tags_with_content(text, ('script', 'noscript'))

# [3] scrapy.utils.response
for text in texts:
    _noscript_re.sub(u'', _script_re.sub(u'', text))
""", setup="""
# [1]
#from scrapy.downloadermiddlewares.ajaxcrawl import _has_ajaxcrawlable_meta
# [2]
#from w3lib.html import remove_tags_with_content
# [3]
from scrapy.utils.response import _noscript_re, _script_re
""")

print timed
```
timings for 1000000 (timeit.default_number) loops:
```
[1] on master branch (scrapy.utils.response)
42.3605451584
[1] on this PR branch (w3lib)
54.5658538342

[2] w3lib.html
33.3376588821

[3] scrapy.utils.response
13.8655879498
```
So performance looks quite a bit worse without the regex compilation.